app/presenters/charge_presenter.rb

```diff
@@ -0,0 +1,15 @@
+class ChargePresenter < BasePresenter
+  # TODO: some methods produce mock data for now
+
+  def offence_name
+    super || id
+  end
+
+  def offence_class
+    'H'
+  end
+
+  def offence_dates
+    super.pluck(:date).compact
```

Comment:
I was wondering if we could remove `nil` dates on save rather than have to `compact` them like this, happy either way though really

Reply:
This really is just a precaution, as we initialise records (charge and associated charge_dates) with nil values for the provider to populate, there is an edge case where the provider might get to the offences page, do not enter nothing, and get back or use the browser history to go back to the "offences summary page" where this offence will contain nil values and blow up (thus this compact really).

Another scenario is in the offence page, without entering anything, the provider can use "Save and come back later" leaving the current offence with nil values.

These are not scenarios that will be common. So is just protecting from nil values. Instead of a compact we can do with a query, like `super.where.not(date: nil)` instead. I used compact as is easier on the test mocks 😄

Reply:
Actually it may be we completely filter out incomplete (as in not "valid" offences -- no name or no dates) from the summary page and that will solve the edge cases. I'm working on the `destroy` action so I can look into that for my next PR.